bitcoin/src/bip32.rs

```diff
@@ -699,6 +713,23 @@ impl Xpriv {
             return Err(Error::UnknownVersion([b0, b1, b2, b3]));
         };
 
+        // Check that byte 45 is 0 as required by BIP-32 for private keys
+        if data[45] != 0 {
+            return Err(Error::InvalidPrivateKeyPrefix);
+        }
+
+        // For master keys (depth=0), both parent fingerprint and child number must be zero
+        let depth = data[4];
+        if depth == 0 {
+            if !data[5..9].iter().all(|&b| b == 0) {
+                return Err(Error::NonZeroParentFingerprintForMasterKey);
+            }
+
+            if !data[9..13].iter().all(|&b| b == 0) {
+                return Err(Error::NonZeroChildNumberForMasterKey);
+            }
+        }
```

Comment:
As I understand it, these should also apply to xpubs, not just xprivs.

Reply:
Ah, yes, good catch!

We should just bring all of https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#test-vector-5 into our unit tests. (Arguably all the test vectors, but that one in particular would've caught this.)

Reply:
Yeah, that's what I'm working on.